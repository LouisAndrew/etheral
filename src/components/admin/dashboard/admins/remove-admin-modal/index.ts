
import { RemoveAdminModal } from './remove-admin-modal';

export default RemoveAdminModal;