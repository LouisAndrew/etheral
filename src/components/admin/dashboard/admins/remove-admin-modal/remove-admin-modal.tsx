import React from 'react';

import { Box, Flex, Button, Heading } from 'rebass';

type Props = {
    toRemoveAdmin: string;
    yes: () => void;
    no: () => void;
};

const RemoveAdminModal: React.FC<Props> = ({ toRemoveAdmin, yes, no }) => {
    return (
        <Box bg="#fff" p={[4]} width={[512]}>
            <Heading mb={[3]} fontSize={3}>
                Are you sure you want to remove {toRemoveAdmin} from etheral
                admin?
            </Heading>
            <Flex>
                <Button width="100%" mr={3} onClick={yes}>
                    Yes
                </Button>
                <Button width="100%" bg="misc.discount" onClick={no}>
                    No
                </Button>
            </Flex>
        </Box>
    );
};

export { RemoveAdminModal };
