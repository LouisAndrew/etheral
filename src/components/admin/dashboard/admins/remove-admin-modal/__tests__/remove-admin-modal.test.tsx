import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { RemoveAdminModal } from '../remove-admin-modal';

describe('RemoveAdminModal', () => {
    const Element = (
        <RemoveAdminModal
            yes={jest.fn(() => {})}
            no={jest.fn(() => {})}
            toRemoveAdmin=""
        />
    );

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
