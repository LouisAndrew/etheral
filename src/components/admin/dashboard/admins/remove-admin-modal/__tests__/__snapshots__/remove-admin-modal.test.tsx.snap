// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveAdminModal matches snapshot 1`] = `
<div
  className="css-w99zwe"
>
  <h2
    className="css-1l4uv3m"
  >
    Are you sure you want to remove 
    
     from etheral admin?
  </h2>
  <div
    className="css-4cffwv"
  >
    <button
      className="css-o6xwpa"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="css-w7s90k"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</div>
`;
